src/embed/ts-embed.ts

```diff
@@ -1047,9 +1047,25 @@ export class TsEmbed {
         if (this.isRendered) {
             logger.warn('Please register event handlers before calling render');
         }
-        const callbacks = this.eventHandlerMap.get(messageType) || [];
-        callbacks.push({ options, callback });
-        this.eventHandlerMap.set(messageType, callbacks);
+        
+        if (messageType === EmbedEvent.ExitPresentMode) {
```

Comment:
Let's not add this here. We should have a different location where we handle all the events coming from the scaligent

Reply:
Refer `src/utils/processData.ts` processEventData